service/src/main/kotlin/fi/espoo/evaka/vasu/VasuController.kt

```diff
@@ -150,18 +153,23 @@ class VasuController(
         @RequestBody body: EditFollowupEntryRequest
     ) {
         Audit.VasuDocumentEditFollowupEntry.log(id, entryId)
-        accessControl.requirePermissionFor(user, Action.VasuDocument.UPDATE, id)
+        accessControl.requirePermissionFor(user, Action.VasuDocument.UPDATE_OWN_FOLLOWUP_ENTRY, id)
         db.transaction { tx ->
             val vasu = tx.getVasuDocumentMaster(id) ?: throw NotFound("vasu $id not found")
             val entry = findFollowupEntryWithId(vasu, entryId) ?: throw NotFound("Entry with $entryId not found")
 
-            if (entry.authorId?.raw  != user.id) {
+            val updateAnyAllowedBy = Action.VasuDocument.UPDATE_ANY_FOLLOWUP_ENTRY.defaultRoles().toTypedArray()
+            val userCanUpdateAnyEntry = user.hasOneOfRoles(*updateAnyAllowedBy)
+
+            if (entry.authorId != user.id && !userCanUpdateAnyEntry) {
```

Comment:
Okay, you changed this here. It would still be ideal if the actual check was done in the `accessControl` side instead of the controller.